inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/gemm_mmad_int8.cl

```diff
@@ -0,0 +1,544 @@
+    for (uint i = 0; i < SUB_GROUP_SIZE; i++) {
+        tile_input2[i] = TO_ACTIVATION_TYPE(input2[batch_offset_input2 + (output_y_tile * TILE_SIZE_M + i) * OUTPUT_SIZE_X + \
+                                                   output_x_tile * TILE_SIZE_N + lid]);
+    }
+#endif // OUTPUT_LEFTOVERS_M || OUTPUT_LEFTOVERS_N
+#endif // INPUT2_TYPE
+
+    ACCUMULATOR_TYPE_VEC tile_output = (ACCUMULATOR_TYPE_VEC)(ACCUMULATOR_VAL_ZERO);
+
+#if !TRANSPOSE_INPUT0
+    const uint K_BLOCK_NUM = (INPUT0_SIZE_X - 1) / TILE_SIZE_K + 1;
+    const uint K_SIZE = INPUT0_SIZE_X;
+#else // !TRANSPOSE_INPUT0
+    const uint K_BLOCK_NUM = (INPUT0_SIZE_Y - 1) / TILE_SIZE_K + 1;
+    const uint K_SIZE = INPUT0_SIZE_Y;
+#endif // !TRANSPOSE_INPUT0
+
+    for (uint k = 0; k < K_BLOCK_NUM; k++) {
+#if !TRANSPOSE_INPUT1
+        MAKE_VECTOR_TYPE(INPUT1_TYPE, PACK_SIZE) temp_input1[SUB_GROUP_SIZE];
+
+        for (uint i = 0; i < SUB_GROUP_SIZE; i++) {
+            const uint common_input1_offset = batch_offset_input1 + (k * TILE_SIZE_K + i * PACK_SIZE) * INPUT1_SIZE_X + output_x_tile * TILE_SIZE_N;
+
+#if OUTPUT_LEFTOVERS_N || OUTPUT_LEFTOVERS_K
+            const uint cur_n = output_x_tile * TILE_SIZE_N + lid;
+            const uint cur_k = k * TILE_SIZE_K + i * PACK_SIZE;
+
+            temp_input1[i] = 0;
+
+            if (cur_n < OUTPUT_SIZE_X) {
+                if (cur_k + 3 < K_SIZE) {
+                    temp_input1[i].s0 = input1[common_input1_offset + lid];
+                    temp_input1[i].s1 = input1[common_input1_offset + INPUT1_SIZE_X + lid];
+                    temp_input1[i].s2 = input1[common_input1_offset + INPUT1_SIZE_X * 2 + lid];
+                    temp_input1[i].s3 = input1[common_input1_offset + INPUT1_SIZE_X * 3 + lid];
+                } else if (cur_k + 2 < K_SIZE) {
+                    temp_input1[i].s0 = input1[common_input1_offset + lid];
+                    temp_input1[i].s1 = input1[common_input1_offset + INPUT1_SIZE_X + lid];
+                    temp_input1[i].s2 = input1[common_input1_offset + INPUT1_SIZE_X * 2 + lid];
+                } else if (cur_k + 1 < K_SIZE) {
+                    temp_input1[i].s0 = input1[common_input1_offset + lid];
+                    temp_input1[i].s1 = input1[common_input1_offset + INPUT1_SIZE_X + lid];
+                } else if (cur_k < K_SIZE) {
+                    temp_input1[i].s0 = input1[common_input1_offset + lid];
+                }
+            }
+#else // OUTPUT_LEFTOVERS_N || OUTPUT_LEFTOVERS_K
+            temp_input1[i].s0 = input1[common_input1_offset + lid];
+            temp_input1[i].s1 = input1[common_input1_offset + INPUT1_SIZE_X + lid];
+            temp_input1[i].s2 = input1[common_input1_offset + 2 * INPUT1_SIZE_X + lid];
+            temp_input1[i].s3 = input1[common_input1_offset + 3 * INPUT1_SIZE_X + lid];
+#endif // OUTPUT_LEFTOVERS_N || OUTPUT_LEFTOVERS_K
+
+            tile_input1[i] = AS_TYPE(PACKED_INPUT1_TYPE, temp_input1[i]);
+        }
```

Comment:
Maybe create two inline functions for calculating `common_input1_offset` and offset in input1, such as: 
```c++
inline uint FUNC(get_common_input1_offset(uint batch_offset_input1, uint k, uint i, uint output_x_tile, uint lid)
{
#if !TRANSPOSE_INPUT1
    return batch_offset_input1 + (k * TILE_SIZE_K + i * PACK_SIZE) * INPUT1_SIZE_X + output_x_tile * TILE_SIZE_N;
#else
    return batch_offset_input1 + (output_x_tile * TILE_SIZE_N + lid) * INPUT1_SIZE_X + k * TILE_SIZE_K + i * PACK_SIZE;
#endif
}
```
and
```c++
inline uint FUNC(get_current_input1_offset)(uint common_input1_offset, uint i, uint lid)
{
#if !TRANSPOSE_INPUT1
    return common_input1_offset + INPUT1_SIZE_X * i + lid;
#else
    return common_input1_offset + i;
#endif
}
```

In this case it should be possible to decrease code duplicating and remove `#if !TRANSPOSE_INPUT1`.